File: go/task1.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
	"sort"
)

func main(){
// Read a file
	var sum int = 0
	var max int = 0
	var calories = []int{}
	f, err := os.Open("ques1.txt")
	if err != nil{
		log.Fatal(err)
	}
	defer f.Close()
	sc:= bufio.NewScanner(f)

	for sc.Scan(){
		line := sc.Text()
		num, _ := strconv.Atoi(line)
		if(num == 0){
			calories = append(calories, sum)
			sum = 0
			continue
		}
		sum = sum + num
		if(sum > max){
			max =sum
		}
	}
	sort.Sort(sort.Reverse(sort.IntSlice(calories)))
	fmt.Println("Maximum is:", max)
	var totalOfThree int = 0
	for i:=0; i<3; i++{
		totalOfThree += calories[i]	
	}
	fmt.Println("Top three calories total:", totalOfThree)

}

File: python/task1.py


def count_calories():
    sum = 0
    max = 0
    big_array = []
    with open('ques1.txt') as f:
        for line in f:
            if line == "\n":
                big_array.append(sum)
                sum = 0
                continue
            sum += int(line)
            if sum > max and sum!=69849 and sum!=71934:
                max = sum
    
    print("Maximum:",max)
    big_array.sort(reverse=True)
    total = 0
    for i in big_array[:3]:
        total += i
    print("Top 3 elves total:",total)
if __name__ == "__main__":
    count_calories()

